certora/specs/SocialRecoveryModule.spec

```diff
@@ -466,3 +466,27 @@ rule invalidatingNonceInRecovery(env e, address guardian, address[] newOwners, u
     bool isReverted = lastReverted;
     assert success => isReverted;
 }
+
+// This rule verifies that Recovery can be finalized after the delay period.
+// This rule requires other conditions to be met as well:
+// - The recovery request should be initiated.
+// - No ether should be sent with the transaction.
+// - New owner should not be a guardian.
+// - Existing Safe owner count should be more than zero.
+rule finalizeRecoveryAlwaysPossible(env e) {
+    uint64 executeAfter = currentContract.recoveryRequests[safeContract].executeAfter;
+    require currentContract.walletsNonces[safeContract] > 0;
```

Comment:
Why is this needed? Is there special meaning to `currentContract.walletsNonces[safeContract] == 0`?

**Edit**: `walletsNonces` can't be 0 for an initialized recovery. Suggested an edit to the comments so this is a bit clearer.

Reply:
This `require` is needed because prover reports a counter example where recovery reverts when the nonce is zero.

Reply:
> prover reports a counter example where recovery reverts when the nonce is zero.


I meant more as in "why" does the prover find a counter example. I found it in the contracts, and it is because once the recovery is initiated, the nonce is incremented, so it is impossible to have `walletsNonce[safe] == 0` before a recovery has initiated. This wasn't super clear to me without reading the code (and why I suggested the edits to the documentation to clarify this bit).